internal/controllers/ipaddressclaim.go

```diff
@@ -160,17 +160,27 @@ func (r *IPAddressClaimReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
-	cluster, err := clusterutil.GetClusterFromMetadata(ctx, r.Client, claim.ObjectMeta)
-	if err == nil {
+	if _, ok := claim.GetLabels()[clusterv1.ClusterNameLabel]; ok {
+		cluster, err := clusterutil.GetClusterFromMetadata(ctx, r.Client, claim.ObjectMeta)
+		if err != nil && !apierrors.IsNotFound(err) {
+			log.Error(err, "error fetching cluster linked to IPAddressClaim")
+			return ctrl.Result{}, err
+		}
+
+		if err != nil && apierrors.IsNotFound(err) {
+			log.Info("IPAddressClaim linked to a cluster that is not found, unable to determine cluster's paused state, skipping reconciliation")
+			return ctrl.Result{}, nil
+		}
```

Comment:
What do you think about this instead?

```suggestion
		if err != nil {
		  if apierrors.IsNotFound(err) {
			  log.Info("IPAddressClaim linked to a cluster that is not found, unable to determine cluster's paused state, skipping reconciliation")
			  return ctrl.Result{}, nil
		  }
			log.Error(err, "error fetching cluster linked to IPAddressClaim")
			return ctrl.Result{}, err
		}
```

Reply:
Sure, I like it like that too. Resolved.